File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/post/dao/PostRepositoryFilter.java
package io.twogether.nbe_5_7_2_02team.post.dao;

import io.twogether.nbe_5_7_2_02team.post.domain.RecruitmentStatus;
import io.twogether.nbe_5_7_2_02team.post.dto.response.PostGetResponse.PostGetResult;

import java.util.List;

public interface PostRepositoryFilter {
    List<PostGetResult> findPostsByMemberId(Long memberId, Long lastPostId, Integer limit);

    List<PostGetResult> findFilteredPosts(
            Long memberId,
            Long lastPostId,
            Integer limit,
            RecruitmentStatus recruitmentStatus,
            Boolean isFollowing,
            List<String> tags);
}

File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/post/dao/PostRepositoryFilter.kt
package io.twogether.nbe_5_7_2_02team.post.dao

import io.twogether.nbe_5_7_2_02team.post.domain.RecruitmentStatus
import io.twogether.nbe_5_7_2_02team.post.dto.response.PostGetResult

interface PostRepositoryFilter {
    fun findPostsByMemberId(
        memberId: Long,
        lastPostId: Long?,
        limit: Int
    ): List<PostGetResult>

    fun findFilteredPosts(
        memberId: Long?,
        lastPostId: Long?,
        limit: Int,
        recruitmentStatus: RecruitmentStatus,
        isFollowing: Boolean?,
        tags: List<String>
    ): List<PostGetResult>
}
